resources/providers/gcplib/inventory/provider.go

```diff
@@ -176,10 +186,13 @@ func (p *Provider) ListAllAssetTypesByName(assetTypes []string) ([]*ExtendedGcpA
 	assets = append(append(assets, resourceAssets...), policyAssets...)
 	mergedAssets := mergeAssetContentType(assets)
 	extendedAssets := extendAssets(p.ctx, p.crm, p.crmCache, mergedAssets)
+	// Enrich network assets with dns policy
+	p.EnrichNetworkAssets(extendedAssets)
```

Comment:
consider moving `p.EnrichNetworkAssets` to a local function and not a method on the provider, as it is meant to be called from within a provider's method itself, similar to `mergeAssetContentType` and `extendAssets`, unlike `ListMonitoringAssets` or `ListAssetTypesByName`